### PR TITLE
feature: JS for dashboard wide filters

### PR DESCRIPTION
**Pull request by gabrielgiroe1**
# Description
Add JavaScript for: [dashboard-wide filters](https://github.com/avo-hq/avo-dashboards/pull/36)

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

**Review by adrianthedev — CHANGES_REQUESTED** [review]
Let's move this into the existing dashboads cards controller

**Review by adrianthedev — CHANGES_REQUESTED** [review]
I made a mistake with the last decision to move it into the dashboard_card controller.

The mistake is that we are attaching this controller to elements that don't need to be refreshed (the previous behavior).

So, please let's move this into a `card_filters_controller`.

Sorry for the back and forth

File: app/javascript/js/controllers/dashboard_card_controller.js [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+
+function updateQueryParam(url, key, value) {
+  const uri = new URI(url)
+  uri.setQuery(key, value)
+
+  return uri.toString()
+}
+
+function getDateParam(url) {
+  const queryString = url.split('?')[1]
```
Comment (adrianthedev) [inline app/javascript/js/controllers/dashboard_card_controller.js:12]: Why are we splitting this as a string?
Can't we use `URI` as well to pick up that param?

Please let's do that.

Comment (gabrielgiroe1) [reply app/javascript/js/controllers/dashboard_card_controller.js]: Yes, sure! I will update it using URI.

**Review by adrianthedev — APPROVED** [review]

**Pull request merged**